.github/workflows/ci.yml

```diff
@@ -37,50 +67,147 @@ jobs:
     steps:
       - uses: actions/checkout@v1
       - uses: actions-rs/toolchain@v1
+        id: toolchain
         with:
+          profile: minimal
           toolchain: stable
           components: clippy
-      - uses: actions-rs/cargo@v1
+      - name: Chown cargo registry because of caching
+        run: sudo chown -R runner ~/.cargo/registry
+      - name: Generate lockfile
+        uses: actions-rs/cargo@v1
+        with:
+          command: generate-lockfile
+      - name: Cache registry
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/registry
+          key: registry-${{ hashFiles('**/Cargo.lock') }}
+      - name: Cache target
+        id: cache-target
+        uses: actions/cache@v1
+        with:
+          path: target
+          key: check-target-${{ steps.toolchain.outputs.rustc_hash }}-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Maybe `clippy-target`?

Reply:
Maybe.
I suppose that clippy uses `check` not `build` inside (so files in `target` are the same for `check` and `clippy` commands).
But maybe it would be better just use a diiferent key. I will make changes.

Reply:
Done